### PR TITLE
173 new appointment

### PR DESCRIPTION
**Pull request by yahgwai**

### CODE REVIEW

File: src/dataEntities/appointment.ts [hunk 1]
```diff
@@ -1,149 +1,328 @@
...
+            eventArgs: appointment.eventArgs,
+            postCondition: appointment.postCondition,
+            paymentHash: appointment.paymentHash
+        };
     }
 
     /**
-     * A combination of the state locator and the nonce for this state update
+     * The hash provided for free the access
      */
```
Comment (bigspider) [inline src/dataEntities/appointment.ts]: Add a more descriptive comment, perhaps.

File: src/dataEntities/appointment.ts [hunk 2]
```diff
@@ -1,149 +1,328 @@
...
-            ...this,
-            startBlock: this.startBlock,
-            endBlock: this.endBlock
+            address: this.contractAddress,
+            topics
         };
     }
+    /**
+     * Th packed representation for the
+     */
```
Comment (bigspider) [inline src/dataEntities/appointment.ts]: Incomplete comment.

File: src/dataEntities/appointment.ts [hunk 3]
```diff
@@ -1,149 +1,328 @@
...
+                this.refund,
+                this.gas,
+                this.mode,
+                ethers.utils.toUtf8Bytes(this.eventABI),
+                this.eventArgs,
+                this.postCondition,
+                this.paymentHash
+            ]
+        );
+    }
```
Comment (bigspider) [inline src/dataEntities/appointment.ts]: If ethers.js doesn't have a way of encoding lists of arguments as arrays of
```
{
 type: "<>",
 value: <>
}
```
or something similar, perhaps we should write one!

File: src/dataEntities/errors.ts [hunk 4]
```diff
@@ -1,3 +1,5 @@
+import { JsonRpcProvider } from "ethers/providers";
+
```
Comment (bigspider) [inline src/dataEntities/errors.ts]: Unused import

File: src/dataEntities/appointment.ts (hunk 3 above)
Comment (yahgwai) [reply src/dataEntities/appointment.ts]: Added a helper function called "groupTuples"

File: src/dataEntities/appointment.ts (hunk 2 above)
Comment (yahgwai) [reply src/dataEntities/appointment.ts]: Corrected

File: src/dataEntities/appointment.ts (hunk 1 above)
Comment (yahgwai) [reply src/dataEntities/appointment.ts]: Added

File: src/dataEntities/errors.ts (hunk 4 above)
Comment (yahgwai) [reply src/dataEntities/errors.ts]: Removed

**Pull request merged**